src/boardlib/api/aurora.py

```diff
@@ -391,3 +392,204 @@ def save_climb(
     )
     response.raise_for_status()
     return response.json()
+
+
+def get_bids_logbook(board, token, user_id):
+    sync_results = user_sync(board, token, user_id, tables=["bids"])
+    return sync_results["PUT"]["bids"]
+
+
+
+
+def bids_logbook_entries(board, username, password, db_path=None):
+    login_info = login(board, username, password)
+    raw_entries = get_bids_logbook(board, login_info["token"], login_info["user_id"])
+    
+    for raw_entry in raw_entries:
+        if db_path:
+            climb_name = get_climb_name_from_db(db_path, raw_entry["climb_uuid"])
+        else:
+            climb_name = get_climb_name(board, raw_entry["climb_uuid"])
+        
+        yield {
+            "uuid": raw_entry["uuid"],
+            "user_id": raw_entry["user_id"],
+            "climb_uuid": raw_entry["climb_uuid"],
+            "climb_name": climb_name if climb_name else "Unknown Climb",
+            "angle": raw_entry["angle"],
+            "is_mirror": raw_entry["is_mirror"],
+            "bid_count": raw_entry["bid_count"],
+            "comment": raw_entry["comment"],
+            "climbed_at": raw_entry["climbed_at"],
+            "created_at": raw_entry["created_at"],
+        }
+
+def get_displayed_grade_from_db(database, climb_uuid, angle, grades_dict):
+    conn = sqlite3.connect(database)
+    cursor = conn.cursor()
+    cursor.execute(
+        "SELECT display_difficulty FROM climb_stats WHERE climb_uuid = ? AND angle = ?", 
+        (climb_uuid, angle)
+    )
+    row = cursor.fetchone()
+    conn.close()
+    if row:
+        difficulty_value = round(row[0])
+        grade_info = grades_dict.get(difficulty_value, {})
+        return grade_info.get("french_name", "NA")
+    return "NA"
+
+def get_full_logbook_entries(board, username, password, grade_type="font", db_path=None):
```

Comment:
This function feels a bit too large. Could we break it into a few logical chunks using helper functions?

Reply:
I tried to refactor the function, I think it's a bit more clear now.